activesupport/lib/active_support/encrypted_configuration.rb

```diff
@@ -35,7 +35,13 @@ def config
 
     private
       def deep_transform(hash)
-        hash.transform_values { |value| value.is_a?(Hash) ? ActiveSupport::InheritableOptions.new(deep_transform(value)) : value }
```

Comment:
I think the problem here is that if you call `ActiveSupport::InheritableOptions.new(foo: "bar")`, you can't interact with it the way you can with a hash literal. I thought you could.

```ruby
2.7.2 :003 > {foo: "bar"}[:foo]
 => "bar"
2.7.2 :004 > Hash.new(foo: "bar")[:foo]
 => {:foo=>"bar"}
```